### DISCUSSION TITLE
Can I use mitmproxy to monkey with YouTube on a smart tv?

### DISCUSSION THREAD

**Discussion by xrd** [mitmproxy/mitmproxy#5965] [Intercept Everything!]
My kids love YouTube but I am troubled by the content they watch. 

I want to make watching YouTube an awful experience, with stuttering, and failed packets, and lots of rebuffering.

Could I use mitmproxy to degrade the experience of YouTube enough to make them disinterested in it?

For example, could I setup mitmproxy and then set my Smart TV to use the proxy. 

(Unfortunately, I don't even see a proxy setting in my TV, so I wonder if I'll need to setup a special Wireless AP to route traffic through that?)

Then, could mitm intercept traffic, and when it sees a DNS request for youtube.com and an IP is assigned, could it then interfere with packet delivery from that IP such that watching YouTube becomes unbearable?

I don't want to completely disable YouTube (I can modify /etc/hosts if I want to block it entirely without using mitmproxy). I just want to make it unbearable so they watch something else more educational.

**Comment by nurettn** [comment]
Hi, you can run mitmproxy with streaming enabled and intercept but Youtube is a complicated platform. I mean I don't know what would be the behaviour of mitmproxy when you try to playback, leap 5 seconds forward/backwards, etc. It would a bit hard to intercept.

If you want to give it a try, check if your TV has IP settings and give the mitmproxy server IP, if not, as you said you can set up a Wireless AP and route through it.

You may use the condition below in your addon and put a random delay, or delete the response buffer content and try to rebuffer it so that the content will load more slowly.

` if "youtube.com" in flow.request.pretty_url and "videoplayback" in flow.request.pretty_url: ...`

You can also get help from [this blog post](https://ericdraken.com/pfsense-decrypt-ad-traffic/#script) to get some insight.

**Reply by 1bsterz** [reply]
Is it possible to intercept the actual video chunks from a  video and alter the frames within those chunks to make them grayscale?

**Reply by mhils** [reply]
Technically yes, practically that likely requires some significant plumbing that's not easy to pull off.

**Comment by Pactortester** [comment]
That's an interesting idea.💡

**Comment by nmz787-intel** [comment]
I think this write-up might get you closer https://www.linkedin.com/pulse/youtube-content-filtering-spartan-budget-sunil-tg-rvp8c/
